api/v1beta1/types.go

```diff
@@ -305,6 +305,8 @@ type SecurityRule struct {
 	// Source specifies the CIDR or source IP range. Asterix '*' can also be used to match all source IPs. Default tags such as 'VirtualNetwork', 'AzureLoadBalancer' and 'Internet' can also be used. If this is an ingress rule, specifies where network traffic originates from.
 	// +optional
 	Source *string `json:"source,omitempty"`
+	// Sources specifies The CIDR or source IP ranges.
+	Sources []*string `json:"sources,omitempty"`
```

Comment:
Just a clarifying question. Sources is a separate field from Source because the SDK separates it into two fields? I'm a bit confused why that is the case.

Reply:
You are right. 
The Azure SDK provides two ways to specify the _Source_ for a security rule. One being `Source` and other being `Sources` for multiple sources for the same rule. 
We see that when converting CAPZ API to Azure SDK one as here https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/3961/files#diff-f1bf1dc0cddcc772a47a442e0d748e6f41bf6f15a502cc138cad235de2bf2d74R32.

So as to why is that the case?

I would bet convenience? 
At Azure Portal, a user can specify a single Source in a rule or have a rule with multiple sources provided they are comma separated. 
So thinking about it, it may have been that Azure first released the API with Source and then added Sources later on to support comma separated Sources behavior.